proposed/2022/AssetTargetFallback-DependenciesResolution.md

```diff
@@ -0,0 +1,406 @@
+
+There a number of issues where this behavior has been observed.
+
+- <https://github.com/NuGet/Home/issues/6522>
+- <https://github.com/NuGet/Home/issues/5871>
+- <https://github.com/NuGet/Home/issues/7263>
+- <https://github.com/NuGet/Home/issues/2315>
+- <https://github.com/NuGet/Home/issues/1661>
+
+The possibility for inconsistency is multiplied with AssetTargetFallback continuing the independent selection.
+However, in practice, the packages that would lead to this type of inconsistency are misauthored.
+
+## Rationale and alternatives
+
+### Do Nothing
+
+The fallback target frameworks are a best effort compatibility mode. There is no guarantee that the assets selected with the fallback. It is reasonable that the consumers need to go the extra mile to enable this scenario.
+The road to success here is evolving the package authoring ecosystem to add support for cross platform scenarios in their packages where appropriate.
+There have been important but not a significant number of issues reported related to this.
+
+**Risks**
+
+- Evolving the package eco system will be a slow transition. Many authors might not be willing to add support for cross platform scenarios.
+Often times these packages might not be maintained. Other times, the consumers take dependencies on such packages for niche scenarios, where the package is guaranteed to work correctly.
+- While not the most pressing issue right now, with more scenarios enabled on .NET Core 3.0, the number of customers hitting this is expected to increase.
+- The workaround can be laborious if the package graph is large and all of the packages there are .NET Framework based.
+
+### Synchronize the selection - Dependencies are the source of truth
+
+The idea here is we extend the promise of selecting assets globally to the dependencies group as well.
+In the well authored package scenario, where each framework supported by the assets has it's own dedicated dependencies group, the source of truth is irrelevant.
+For example the NuGet.PackageManagement scenario will easily work.
+The challenge is when the answers to these questions is different.
+
+**Risks**
+
+We would not consider these packages well authored today, but that has not always been our guidance. Furthermore many packages are authored with a nuspec, where not obvious issues like these can easily happen.
+
+Say we have `netcoreapp3.0` project and a package like below:
+
+```text
+lib/net472/a.dll
+lib/netstandard2.0/a.dll
+```
+
+```xml
+    <dependencies>
+      <group targetFramework=".NETFramework4.7.2">
+        <dependency id="X" version="2.1.2" exclude="Build,Analyzers" />
+      </group>
+    </dependencies>
+```
+
+The package author here might have wanted to suggest that their package has no dependencies for `netstandard2.0`, but NuGet does not interpret it such.
+Today if you pack a project that multi targets, NuGet will put an empty dependency group for the target frameworks that do not have any package references.

```

Comment:
```suggestion
The package author here might have wanted to suggest that their package has no dependencies for `netstandard2.0`, but NuGet does not interpret it such.
Today if you pack a project that multi targets, NuGet will put an empty dependency group for the target frameworks that do not have any package references.
Pack warning NU5128 warn package authors of this problem when their pack scenario is different to the default NuGet pack targets output.
```

Reply:
this is talking about dependecyGroup, not dependencies or anything like that. 

But that usggestion was super helpful in the other part.